test/cpu_reduce_util.hpp

```diff
@@ -74,20 +74,19 @@ static inline bool float_equal_one(T x)
 template <>
 inline bool float_equal_one<float>(float x)
 {
-    return std::isfinite(x) and x <= 1.0f and x >= 1.0f;
+    return std::isfinite(x) and x == 1.0f;
```

Comment:
BTW isfinite is not necessary as well, esp. if you care about performance. Up to you.

Reply:
Just go to the simplest codes and I have verified it works correctly on host side: 
1)  operator ==  works for both float/double. It is able to regard two zero values as equal when one of them is 0x00000000, the other is 0x80000000 
2) half_float::half also implemented the correct overload for operator ==